### ISSUE TITLE
docs and wokwi_test GHAs fail when submodules are used

### ISSUE THREAD

**Issue by algofoogle** [TinyTapeout/tt03p5-submission-template#1]
# tl;dr

Add `submodules: recursive` to the initial checkout steps of the `docs` and `wokwi_test` GitHub Actions. This will match them to how the `gds` Action works, and everything will be sweet :)

# Detail

The `gds` GitHub Action fetches submodules recursively [using `submodules: recursive`](https://github.com/TinyTapeout/tt03p5-submission-template/blob/main/.github/workflows/gds.yaml#L24)

The `docs` and `wokwi_test` GHAs, however, [lack this](https://github.com/TinyTapeout/tt03p5-submission-template/blob/37d567519aed9a28a86073d921b8297ec26ae27f/.github/workflows/docs.yaml#L15), which leads to a failure where it inherently tries to find Verilog code that happens to belong to a submodule.

This appears as an error about a missing Verilog file during [the `tt_tool.py --check-docs` step](https://github.com/TinyTapeout/tt03p5-submission-template/blob/37d567519aed9a28a86073d921b8297ec26ae27f/.github/workflows/docs.yaml#L38).

Example failure: https://github.com/algofoogle/tt03p5-solo-squash/actions/runs/5144949991/jobs/9261910775#step:7:13:
```
Run ./tt/tt_tool.py --check-docs
  ./tt/tt_tool.py --check-docs
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.10.11/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.10.11/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.10.11/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.11/x64/lib
2023-06-01 13:28:36,663 - project    - ERROR    - solo_squash/src/solo_squash.v doesn't exist in the repo
```
....in this case, `solo_squash` is the submodule inside the TT `src/` dir. The `gds` GHA retrieves this and passes, but the `docs` and `wokwi_test` Actions fail as seen above. I suspect those particular Actions don't even use those files (but validating that they are there is a built-in part of `project.py`).

**Comment by algofoogle** [comment]
Simply [adding the directive](https://github.com/algofoogle/tt03p5-solo-squash/commit/2eda11c5c5858182e25d491706a16c2312f894e9) seems to do the trick, but I **assume** will need to be part of your master workflow/tools...?

**Comment by urish** [comment]
Thanks for reporting! Fixed for TT04
